bundles/framework/layerselector2/Flyout.js

```diff
@@ -104,28 +99,40 @@ Oskari.clazz.define('Oskari.mapframework.bundle.layerselector2.Flyout',
             elParent = this.container.parentElement.parentElement;
             elId = jQuery(elParent).find('.oskari-flyouttoolbar .oskari-flyouttools .oskari-flyouttool-close');
             elId.attr('id', 'oskari_layerselector2_flyout_oskari_flyouttool_close');
-
-            var buttons = me.layerlistService.getLayerlistFilterButton();
-            Object.keys(buttons).forEach(function(key) {
-                var button = buttons[key];
-                me.addFilterTool(button.text, button.tooltip, button.cls.active, button.cls.deactive, button.id);
-            });
+            
+            me.createFilterButtons();
         },
-
         /**
-         * Adds default filter buttons.
-         * @method  @private addDefaultFilters
+         * Create filterbuttons for each active filter
+         * @method  @public createFilterButtons
          */
-        addDefaultFilters: function() {
+        createFilterButtons: function () {
             var me = this;
+            this.layerTabs.forEach( function ( tab ) {
+                var filterButton = Oskari.clazz.create("Oskari.layerselector2.view.FilterButtons", tab.getTabPanel().getContainer().find('.layerselector2-layer-filter') );
+                me.filterComponents.push(filterButton);
 
-            // Add newest filter
-            me.addNewestFilter();
-
-            // Add featuredata filter
-            me.addFeaturedataFilter();
+                filterButton.on('FilterActivate', function (currentFilter) {
+                    me._currentFilter = currentFilter;
+                    me.populateLayers();
+                });
+                
+            });
         },
+        setActiveFilter: function () {
+            var me = this;
```

Comment:
Shouldn't this take me._currentFilter as parameter instead of having a "set" method without any parameters?